.github/workflows/api-build-deploy.yaml

```diff
@@ -0,0 +1,115 @@
+name: Build and Deploy API
+
+on:
+  pull_request:
+    types: [synchronize]
+  push:
+    branches: [ main ]
+    paths:
+      - "api/**"
+  workflow_dispatch:
+
+jobs:
+  build-tag-push:
+    name: Build, Tag, Push to Artifactory
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Log In to Docker
+        run: |
+          docker login ${{ secrets.ARTIFACTORY_URL }} -u ${{ secrets.ARTIFACTORY_USERNAME }} -p ${{ secrets.ARTIFACTORY_PASSWORD }}
+
+      - name: Copy env.template
+        run: |
+          cp .env.template .env
+      
+      - name: Build API Image
+        run: |
+          docker-compose build spr-api
+
+      - name: List Docker Images
+        run: |
+          docker images
+        
+      - name: Tag Image
+        run: |
+          docker tag citz-imb-staff-purchasing-reimbursement_spr-api ${{ secrets.ARTIFACTORY_URL }}/${{ secrets.IMAGE_REPOSITORY }}/spr-api:latest
+      
+      - name: Push Image
+        run: |
+          docker push ${{ secrets.ARTIFACTORY_URL }}/${{ secrets.IMAGE_REPOSITORY }}/spr-api:latest
+  
+  deploy:
+    name: Deploy to OpenShift
+    runs-on: ubuntu-latest
+    needs: [ build-tag-push ]
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v3
+
+      - name: Login to Openshift
+        uses: redhat-actions/oc-login@v1
+        env:
+          OPENSHIFT_USER: github-actions
+          OPENSHIFT_PROJECT: ${{ secrets.OPENSHIFT_PROJECT }}
+        with: 
+          openshift_server_url: ${{ secrets.OPENSHIFT_SERVER_URL }}
+          openshift_token: ${{ secrets.OPENSHIFT_TOKEN }}
+          namespace: ${{ secrets.OPENSHIFT_PROJECT }}
+
+      - name: Deploy from DeploymentConfig
+        env:
+          OPENSHIFT_PROJECT: ${{ secrets.OPENSHIFT_PROJECT }}
+          APPLICATION_NAME: staff-purchase-reimbursement
+          CONTAINER_NAME: spr-api
+          DEPLOYMENT_CONFIG: api-deployment-config.yaml
+          IMAGE_REPOSITORY: ${{ secrets.IMAGE_REPOSITORY }}
+        run: |
+          chmod +x ./.github/helpers/deploy.sh &&
+          ./.github/helpers/deploy.sh
```

Comment:
```suggestion
          ./.github/helpers/deploy.sh
```

Reply:
run the command I gave you in teams to perm add execute permissions to your helper scripts

Reply:
Done